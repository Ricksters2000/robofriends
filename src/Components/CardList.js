import React from 'react';
import Card from './Card';

const CardList = ({robots}) => {
    //used to return an error with the help of error boundries (useful for displaying user-friendly errors)
    // if(true) {
    //     throw new Error('has recieved an error');
    // }

    const cardComponent = robots.map((user, i) => <Card key={i} name={robots[i].name} id={robots[i].id} email={robots[i].email} />);
    
    return(
        <div>
            {cardComponent}
        </div>
    );
}

export default CardList;