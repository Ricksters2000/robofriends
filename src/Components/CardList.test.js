import React from 'react';
import {shallow, mount, render} from 'enzyme';
import CardList from './CardList';

it('expect to render Card component', () => {
    const mockRobots = [
        {
            id: 1,
            name: 'something',
            username: 's',
            email: 's@gmail.com'
        }
    ]
    expect(shallow(<CardList robots={mockRobots} />)).toMatchSnapshot();
})
