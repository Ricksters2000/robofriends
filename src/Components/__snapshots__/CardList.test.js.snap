// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render Card component 1`] = `
<div>
  <Card
    email="s@gmail.com"
    id={1}
    key="0"
    name="something"
  />
</div>
`;
